micrometer-core/src/main/java/io/micrometer/core/aop/CountedAspect.java

```diff
@@ -145,20 +145,17 @@ private void recordCompletionResult(ProceedingJoinPoint pjp, Counted counted, Th
     }
 
     private void record(ProceedingJoinPoint pjp, Counted counted, String exception, String result) {
-        counter(pjp, counted)
-                .tag(EXCEPTION_TAG, exception)
-                .tag(RESULT_TAG, result)
-                .tags(counted.extraTags())
-                .register(meterRegistry)
-                .increment();
-    }
-
-    private Counter.Builder counter(ProceedingJoinPoint pjp, Counted counted) {
-        Counter.Builder builder = Counter.builder(counted.value()).tags(tagsBasedOnJoinPoint.apply(pjp));
-        String description = counted.description();
-        if (!description.isEmpty()) {
-            builder.description(description);
+        try {
```

Comment:
Why add the try-catch? Is it because the user-provided `tagsBasedOnJoinPoint` function might throw an exception?

Reply:
I opened a separate issue for this: https://github.com/micrometer-metrics/micrometer/issues/5584

Reply:
Yes `tagsBasedOnJoinPoint` may throw an exception and we may want to handle the expectation explicitly.
Do you prefer a fix in the current PR or a separate one based on https://github.com/micrometer-metrics/micrometer/issues/5584?

Reply:
I think it will be better to handle that separately from this PR. I opened #5585 for it. Thanks for bringing it to our attention.